test/test_async_vault.py

```diff
@@ -161,3 +161,12 @@ class TestSettings:
     config._initialized = True
 
     assert await config.asdict() == {"SECRET": 1, "VAULT_ADDR": vault_addr, "VAULT_TOKEN": vault_token}
+
+
+async def test_retry(config, mocker):
+    mocker.patch("requests.adapters.HTTPAdapter.send", side_effect=aiohttp.ClientConnectionError)
```

Comment:
This one is not related to the async vault, it is not used, there should be an equivalent for `aiohttp` (e.g. in `ClientSession` class)